### PR TITLE
Gutenboarding: add header

### PR DESCRIPTION
**Pull request by simison**
Adds a basic header canvas with an example button.

As next steps outside this PR we need to somehow ensure that `.layout__content` and Calypso CSS reset styles don't collide with block editor.

In this PR we introduce a bunch of [Gutenberg stylesheets](https://github.com/WordPress/gutenberg/tree/26795cac83cbcd4a78a2045c444d9d30ff2f0590/assets/stylesheets) in Calypso, but those should be in future imported via `base-styles` package, being worked at https://github.com/WordPress/gutenberg/pull/17883.

## Testing

- Visit `/gutenboarding` in development env and confirm that you see a header that doesn't do anything yet. :-) 
- Confirm that CSS reset isn't leaking to rest of Calypso

### CODE REVIEW

**Comment by chriskmnds** [comment]
Looks like a bit of a structure emerging. How about something like this for the additional content:

```
./assets/
- stylesheets/
- stylesheets/main.scss
- stylesheets/gutenberg/
- stylesheets/gutenberg/_animations.scss 
- stylesheets/gutenberg/_breakpoints.scss
- stylesheets/gutenberg/_colors.scss 
- stylesheets/gutenberg/_mixins.scss
- stylesheets/gutenberg/_variables.scss
- stylesheets/gutenberg/_z-index.scss

./src/
- components/
- components/header/
- components/header/index.tsx
- components/header/style.scss
- components/inserter/
- components/inserter/index.tsx 
```

We can do more with the rest too e.g.

```
./src/
- lib/
- lib/routing
- lib/routing/controller.tsx

./src/
- sections/
- sections/main/
- sections/main/index.tsx {current gutenboarding.tsx}
- sections/create-site/
- sections/create-site/index.tsx
```

Happy to move these if we agree (probably do all in a separate PR I guess).

**Comment by sirreal** [comment]
Rebased and targeted to `master`.

**Comment by simison** [comment]
All Gutenberg core styles here should be imported from possible future `@wordpress/base-styles` package (https://github.com/WordPress/gutenberg/pull/17883) instead of copying them over.

**Comment by ockham** [comment]
With paObgF-BE-p2 in mind, what about replicating `edit-post`'s [directory structure](https://github.com/WordPress/gutenberg/tree/26795cac83cbcd4a78a2045c444d9d30ff2f0590/packages/edit-post/src) more closely? I.e. use a `components/` subfolder for the sidebar etc? (Upstream's `edit-widgets` also [does that](https://github.com/WordPress/gutenberg/tree/26795cac83cbcd4a78a2045c444d9d30ff2f0590/packages/edit-widgets/src) -- hoping to provide some amount of consistency should we ever converge with core's FSE.)

**Comment by simison** [comment]
Valid points about components folder @ockham & @chriskmnds! Updated in https://github.com/Automattic/wp-calypso/pull/36525/commits/a050910074630277b756bfa18458c3a409b3e93a

File: client/gutenboarding/gutenberg-styles/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Gutenberg core stylesheets
+
+From [`WordPress/gutenberg/assets/stylesheets`](https://github.com/WordPress/gutenberg/tree/399e5e143a62a66101285de623cf47fab159bbdb/assets/stylesheets) on Gutenberg v6.6.0.
```
Comment (ockham) [inline client/gutenboarding/gutenberg-styles/README.md]: Should we add a `TODO` to state that we'll replace the copy pasta with `@wordpress/base-styles` once https://github.com/WordPress/gutenberg/pull/17883 is merged and the package has been published?

**Review by ockham — APPROVED** [review]
LGTM; one [minor note](https://github.com/Automattic/wp-calypso/pull/36525/files#r336897678).

(I didn't review the styling closely since it seems to be mostly copied from Gutenberg.)

File: client/gutenboarding/gutenberg-styles/README.md (hunk 1 above)
Comment (simison) [reply client/gutenboarding/gutenberg-styles/README.md]: Good idea!  https://github.com/Automattic/wp-calypso/pull/36525/commits/f935c939426f31dde878b01c135104ff3e5c8851

**Pull request merged**